### ISSUE TITLE
bug(#500 follow-up): dc9e67c does not fix all-NaN at 1280×704 — true root cause is b0 FFN bf16 overflow at long seq

### ISSUE THREAD

**Issue by dahai80** [dahai80/fusion-mlx#503]
## Summary

#500 was closed by dc9e67c (auto-chunk `FUSION_WAN2_ATTN_CHUNK=8192` + fail-visible VAE NaN raise), but **the fix does not resolve the all-NaN at 1280×704×121f**. The chunking mitigation is orthogonal to the true root cause, which this issue identifies precisely.

**True root cause: block-0 FFN output overflows to NaN at long patch-seq (27280), not the self-attention SDPA.**

Verified via in-process boundary probes (monkeypatched `WanAttentionBlock.__call__` / `WanSelfAttention` / `WanCrossAttention` module-exit stats, no mid-graph recompute):

| stage | 512×512 (seq=7936) | 1280×704 (seq=27280) |
|---|---|---|
| b0 self_attn out `y` | 20.1 finite | 19.6 finite ✅ |
| b0 cross_attn out `y` | 0.64 finite | 7.4 finite ✅ |
| **b0 block output** | finite | **NaN (162997)** ❌ |

So b0 self_attn and cross_attn are both finite at 1280×704; NaN is born **between cross_attn exit and block exit** — i.e. the b0 FFN path (`norm2 → fc1 → GELU → fc2 → residual add`). It is resolution/seq dependent: identical b0 weights, similar residual magnitude (~48), only seq differs (7936 vs 27280).

Falsified hypotheses (all probed):
- ❌ bf16 SDPA kernel seq-bug — synthetic `mx.fast.scaled_dot_product_attention` fine to seq=40000
- ❌ RoPE corruption — bounded at both resolutions
- ❌ chunking — `FUSION_WAN2_ATTN_CHUNK=0` (single SDPA) reproduces identical NaN
- ❌ v-projection saturation — `sa.v(x_w)` raw output finite (max 21.875) at b0
- ❌ o-projection saturation — standalone `self.o` (q8) does not saturate even at input am=50000
- ❌ logits/softmax overflow — logits max 21.97 finite

The FFN path (`WanFFN.__call__`: `x_w = x.astype(_linear_dtype(self.fc1)); fc2(act(fc1(x_w)))`) runs the 4× expansion + GELU + contraction in bf16. At seq=27280 the bf16 FFN overflows where fp32 (the official Wan2 residual autocast keeps the stream float32) does not.

## Reproduction

```python
from fusion_mlx.video.wan2.generate import generate_video
generate_video(
    model_dir="~/.fusion-mlx/models/Wan2.2-TI2V-5B-mlx-q8",
    prompt="A cat walking on a beach, cinematic",
    negative_prompt="static, low quality",
    image=None, width=1280, height=704, num_frames=121,
    steps=1, guide_scale=5.0, shift=5.0, seed=2524546589,
    output_path="/tmp/ti2v_repro.mp4",
    output_format="raw", scheduler="unipc", no_compile=True,
)
# → "Denoise step 1/1: 5237568 NaN in latents — aborting"
# → "Wan2 VAE decode produced 327106560 NaN pixels"
```

With `FUSION_WAN2_ATTN_CHUNK=0` (chunking disabled) → identical 5237568 NaN, proving chunking is not the fix.

## Secondary bug in dc9e67c

`attention.py:68` calls `mx.eval(out_chunk)` inside `_sdpa_chunked`, which raises under `mx.compile`:
`Attempting to eval an array during function transformations like compile or vmap`.
Worked around in probes with `no_compile=True`, but this crashes the default compiled path when chunking triggers.

## Proposed fix direction

Run the Wan2 FFN (and/or the full residual stream) in float32 at long seq, matching the official Wan2.1/2.2 autocast which keeps the residual float32. Minimal surgical option: cast the FFN contraction `fc2` input to float32 so its matmul accumulates without bf16 saturation. Open to the maintainer's preferred approach.

## Environment

- fusion-mlx 0.8.21 (commit b10f7e6), MLX 0.31.2
- Model: Wan2.2-TI2V-5B-mlx-q8, dim=3072 ffn=14336 heads=24 layers=30
- Apple Silicon

**Comment by dahai80** [comment]
## Root cause confirmed + fix in PR #504

The true root cause of the all-NaN at 1280x704x121f is **MLX `QuantizedLinear` bf16-accumulator overflow**, not SDPA saturation and not the residual-stream dtype (both prior hypotheses were falsified by instrumentation).

### Mechanism
MLX `QuantizedLinear` with **bfloat16 input** accumulates the per-group dot product in bfloat16. At long seq (27280) the residual stream grows and the q8 linear projections overflow the bf16 accumulator to bf16 max (`1.70141e38`) or NaN, even though the bf16 input itself is finite. The saturation then propagates through the residual and norms -> fp32 overflow -> all-NaN latents. PyTorch fp16/bf16 matmuls accumulate in float32, so the official Wan2 never hits this — it is MLX-q8-specific.

### Confirmed overflow sites (block-by-block probes)
| site | bf16 input | bf16 output |
|------|-----------|-------------|
| `self_attn.o` | am=5.65 (finite) | `1.7e38` saturated |
| `cross_attn.o` | am=1.27 (finite) | nan=106170 |
| `ffn.fc2` (14336 in) | am=6.65 (finite) | nan=106170 |
| q/k/v-proj (later blocks) | residual am~73 | overflow |

### Fix
Cast the input of every q8 linear (q/k/v/o in both attentions + fc1/fc2 in FFN) to `float32` when denoiser token count > `FUSION_WAN2_Q8_FP32_SEQ` (default 16384). fp32 input -> q8 matmul accumulates in fp32 -> finite. Short-seq runs (seq<16384, e.g. 512x512) keep the fast bf16 path unchanged.

### Verified
- 1280x704x121f: GEN OK, BLK0-2 finite
- 512x512x121f: GEN OK (regression clean)
- wan2 unit tests: 34 passed

See PR #504 for the implementation.
